src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -5,86 +5,114 @@
 
 #define THIS ((BossTw*)thisx)
 
+typedef enum {
+    /*  0 */ TWEFF_NONE,
+    /*  1 */ TWEFF_DOT,
+    /*  2 */ TWEFF_2,
+    /*  3 */ TWEFF_3,
+    /*  4 */ TWEFF_RING,
+    /*  5 */ TWEFF_PLYR_FRZ,
+    /*  6 */ TWEFF_FLAME,
+    /*  7 */ TWEFF_MERGEFLAME,
+    /*  8 */ TWEFF_SHLD_BLST,
+    /*  9 */ TWEFF_SHLD_DEFL,
+    /* 10 */ TWEFF_SHLD_HIT
+} TwEffType;
+
+typedef enum {
+    /* 0 */ EFF_ARGS,
+    /* 1 */ EFF_UNKS1,
+    /* 2 */ EFF_WORK_MAX
+} EffectWork;
+
+typedef enum {
+    /* 0 */ EFF_SCALE,
+    /* 1 */ EFF_DIST,
+    /* 2 */ EFF_ROLL,
+    /* 3 */ EFF_YAW,
+    /* 4 */ EFF_FWORK_MAX
+} EffectFWork;
+
+typedef enum {
+    /* 0x00 */ TW_KOTAKE,
+    /* 0x01 */ TW_KOUME,
+    /* 0x02 */ TW_TWINROVA,
+    /* 0x64 */ TW_FIRE_BLAST = 0x64,
+    /* 0x65 */ TW_FIRE_BLAST_GROUND,
+    /* 0x66 */ TW_ICE_BLAST,
+    /* 0x67 */ TW_ICE_BLAST_GROUND,
+    /* 0x68 */ TW_DEATHBALL_KOTAKE,
+    /* 0x69 */ TW_DEATHBALL_KOUME
+} TwinrovaType;
+
+typedef struct {
+    /* 0x0000 */ u8 type;
+    /* 0x0001 */ u8 frame;
+    /* 0x0004 */ Vec3f pos;
+    /* 0x0010 */ Vec3f curSpeed;
+    /* 0x001C */ Vec3f accel;
+    /* 0x0028 */ Color_RGB8 color;
+    /* 0x002C */ s16 alpha;
+    /* 0x002E */ s16 work[EFF_WORK_MAX];
+    /* 0x0034 */ f32 workf[EFF_FWORK_MAX];
+    /* 0x0044 */ Actor* target;
+} BossTwEEffect;
+
 void BossTw_Init(Actor* thisx, GlobalContext* globalCtx);
 void BossTw_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void BossTw_Update(Actor* thisx, GlobalContext* globalCtx);
 void BossTw_Draw(Actor* thisx, GlobalContext* globalCtx);
+void BossTw_TwinrovaDamage(BossTw* this, GlobalContext* globalCtx, u8 arg2);
+void BossTw_TwinrovaSetupFly(BossTw* this, GlobalContext* globalCtx);
+void BossTw_DrawEffects(GlobalContext* globalCtx);
+void BossTw_TwinrovaLaugh(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaFly(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaGetUp(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaSetupGetUp(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaSetupLaugh(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaDoneBlastShoot(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaSetupDoneBlastShoot(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaSetupShootBlast(BossTw* this, GlobalContext* globalCtx);
+void BossTw_SetupTwinrovaChargeBlast(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaArriveAtTarget(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaDeathCS(BossTw* this, GlobalContext* globalCtx);
+void BossTw_TwinrovaIntroCS(BossTw* this, GlobalContext* globalCtx);
+void BossTw_CSWait(BossTw* this, GlobalContext* globalCtx);
```

Comment:
this is more "wait for cs", its not a part of the cutscene

Reply:
my reasoning was to keep all cutscene-related functions consistently named, if this shouldn't be considered part of the cutscene may as well drop the "CS" then (but `BossTw_Wait` is already taken by a function that waits to... setup a cutscene)

`BossTw_CSWait` -> `BossTw_Wait` (since it does litterally nothing but waiting)
`BossTw_Wait` -> `BossTw_WaitMerge` (or WaitPhase2 idk)
?

Reply:
gonna just leave as is for now and it can be revisited in a more detailed docs pass. I dont feel comfortable naming things without knowing the system very well